core/src/apps/cardano/address.py

```diff
@@ -34,6 +40,71 @@
 MAX_ADDRESS_BYTES_LENGTH = 65
 
 
+def validate_address_parameters(parameters: CardanoAddressParametersType) -> None:
+    _validate_address_parameters_fields(parameters)
+
+    if parameters.address_type == CardanoAddressType.BYRON:
+        if not is_byron_path(parameters.address_n):
+            raise INVALID_ADDRESS_PARAMETERS
+    elif parameters.address_type in ADDRESS_TYPES_SHELLEY:
+        if not is_shelley_path(parameters.address_n):
+            raise INVALID_ADDRESS_PARAMETERS
+
+        if parameters.address_type == CardanoAddressType.BASE:
+            _validate_base_address_staking_info(
+                parameters.address_n_staking, parameters.staking_key_hash
+            )
+        elif parameters.address_type == CardanoAddressType.POINTER:
+            if parameters.certificate_pointer is None:
+                raise INVALID_ADDRESS_PARAMETERS
+        elif parameters.address_type == CardanoAddressType.REWARD:
+            if not SCHEMA_STAKING_ANY_ACCOUNT.match(parameters.address_n):
+                raise INVALID_ADDRESS_PARAMETERS
+    else:
+        raise INVALID_ADDRESS_PARAMETERS
+
+
+def _validate_address_parameters_fields(
```

Comment:
I find the name somewhat misleading, like if it was about individual fields. What about `_check_superfluous_fields` or something like that?

Reply:
Maybe `_validate_address_parameters_structure`? `_validate_address_parameters_format`?

Reply:
I've changeed it to `validate_address_parameters_structure` https://github.com/vacuumlabs/trezor-firmware/pull/60/commits/8b08dc8000fc77243a02062c76dfb8752eeee064.